rustls/src/quic.rs

```diff
@@ -561,16 +564,26 @@ impl AsRef<[u8]> for Tag {
 }
 
 /// How a `Tls13CipherSuite` generates `PacketKey`s and `HeaderProtectionKey`s.
-pub(crate) trait Algorithm: Send + Sync {
+pub trait Algorithm: Send + Sync {
+    /// Produce a `PacketKey` encrypter/decrypter for this suite.
+    ///
+    /// `suite` is the entire suite this `Algorithm` appeared in.
+    /// `key` and `iv` is the key material to use.
```

Comment:
Given that the fields of `AeadKey` are private, I'm not sure this trait is practically implementable for foreign types?

Reply:
The expected things for people to do with an `AeadKey` is:

- view the raw bytes, via `impl AsRef<[u8]>`
- move or drop it (has a non-trivial zeroizing `impl Drop`)

I don't think there is any need for construction of these outside the crate.

Reply:
Ahh, yes -- sorry for my confusion.